src/PackagePublisher/Program.cs

```diff
@@ -20,13 +20,12 @@ public static class Program
         private const string ApiKeyEnvironmentVariableName = "APIKey";
         private static CommandOption _packagesDirectory;
         private static CommandOption _feedToUploadTo;
-        private static CommandOption _maxRetryCountOption;
 
         // Do NOT log the key
         private static string _apiKey = null;
         private static int _maxRetryCount = 5;
-        private static int _packagePushTimeoutInSeconds = 90;
-        private static int _semaphoreTimeoutInSeconds = _maxRetryCount * _packagePushTimeoutInSeconds;
+        private static readonly TimeSpan _packagePushTimeout = TimeSpan.FromSeconds(90);
+        private static readonly TimeSpan _semaphoreTimeout = TimeSpan.FromHours(1);
```

Comment:
Yeah, that might be an better solution. Let's go with this for now and we can refactor this code later on.